src/Misc/layoutbin/powershell/Use-VstsTaskScript.ps1

```diff
@@ -0,0 +1,42 @@
+<# 
+A PowerShell script that is used to invoke a VSTS task script. This script is used by the VSTS task runner to invoke the task script.
+This script resplaces some legacy stuff in PowerShell3Handler.cs and turns it into a dedicated function. 
+since it is parameterized it can be signed and trusted for WDAC and CLM.
+This is the original code that has been repalced - 
+@". ([scriptblock]::Create('if ([Console]::InputEncoding -is [Text.UTF8Encoding] -and [Console]::InputEncoding.GetPreamble().Length -ne 0) {{ [Console]::InputEncoding = New-Object Text.UTF8Encoding $false }} if (!$PSHOME) {{ $null = Get-Item -LiteralPath ''variable:PSHOME'' }} else {{ Import-Module -Name ([System.IO.Path]::Combine($PSHOME, ''Modules\Microsoft.PowerShell.Management\Microsoft.PowerShell.Management.psd1'')) ; Import-Module -Name ([System.IO.Path]::Combine($PSHOME, ''Modules\Microsoft.PowerShell.Utility\Microsoft.PowerShell.Utility.psd1'')) }}')) 2>&1 | ForEach-Object {{ Write-Verbose $_.Exception.Message -Verbose }} ; Import-Module -Name '{0}' -ArgumentList @{{ NonInteractive = $true }} -ErrorAction Stop ; $VerbosePreference = '{1}' ; $DebugPreference = '{1}' ; Invoke-VstsTaskScript -ScriptBlock ([scriptblock]::Create('. ''{2}'''))",
+#>
+
+param ( 
+    [Parameter(mandatory = $true)]$name, # Argument 0 
+    [Parameter(mandatory = $true)]$DebugOption, # Argument 1
+    [Parameter(mandatory = $true)]$scriptBlockString	# Argument 2 
```

Comment:
The params names don't seem to match the place it is called from 🙂 
https://github.com/microsoft/azure-pipelines-agent/pull/4502/files#diff-a4b15b8f89712d3eeda4d7a79e9bc8720fff5807c606a63e1782f196f0eb96a8R59

Reply:
@KonstantinTyukalov, I've synced the names up. I have a habit of just using enough characters to disambiguate the name in my scripts, not the best to embed into a product. :)